src/main/resources/dbreplica/migration/V1.0.8__reportable_task_plsql.sql

```diff
@@ -0,0 +1,256 @@
+l_first_assigned_date     DATE;
+l_first_assigned_date_time     TIMESTAMP;
+l_number_of_reassignments      INTEGER := -1;
+l_wait_time_days          INTEGER;
+l_handling_time_days      INTEGER;
+l_processing_time_days    INTEGER;
+l_is_within_sla           TEXT;
+l_wait_time                       INTERVAL;
+l_handling_time                   INTERVAL;
+l_processing_time                 INTERVAL;
+l_due_date_to_completed_diff_time INTERVAL;
+l_state_label             TEXT;
+l_role_category_label     TEXT;
+l_jurisdiction_label      TEXT;
+l_case_type_label         TEXT;
+
+
+task_history_cursor CURSOR FOR
+SELECT task_id,task_name,task_type,due_date_time,state,task_system,security_classification,title,major_priority,
+       minor_priority,assignee,auto_assigned,execution_type_code,work_type,role_category,has_warnings,assignment_expiry,
+       case_id,case_type_id,case_category,case_name,jurisdiction,region,location,business_context,termination_reason,
+       created,updated_by,updated,update_action
+FROM cft_task_db.task_history
+WHERE task_id = l_task_id order by updated;
+
+begin
+
+select not exists(select 1 from cft_task_db.reportable_task where reportable_task.task_id = l_task_id)
+into l_new_task;
+
+OPEN task_history_cursor;
+
+LOOP
+FETCH NEXT FROM task_history_cursor INTO
+    l_task_id,l_task_name,l_task_type,l_due_date_time,l_state,l_task_system,l_security_classification,l_title,l_major_priority,
+    l_minor_priority,l_assignee,l_auto_assigned,l_execution_type_code,l_work_type,l_role_category,l_has_warnings,l_assignment_expiry,
+    l_case_id,l_case_type_id,l_case_category,l_case_name,l_jurisdiction,l_region,l_location,l_business_context,l_termination_reason,
+    l_created,l_updated_by,l_updated,l_update_action;
+
+    -- Exit the loop if no more rows are available
+    EXIT WHEN NOT FOUND;
+
+    l_role_category_label =
+        case l_role_category IS NOT NULL
+            when (l_role_category='LEGAL_OPERATIONS')  then 'Legal Operations'
+            when (l_role_category='CTSC')  then 'CTSC'
+            when (l_role_category='JUDICIAL')  then 'Judicial'
+            when (l_role_category='ADMINISTRATOR')  then 'Admin'
+            when (l_role_category='ADMIN')  then 'Admin'
+            when (l_role_category='')  then 'Blank values'
+            else l_role_category
+        end;
+
+    l_jurisdiction_label =
+        case l_jurisdiction IS NOT NULL
+            when (l_jurisdiction='PRIVATELAW')  then 'Private Law'
+            when (l_jurisdiction='CIVIL')  then 'Civil'
+            when (l_jurisdiction='IA')  then 'Immigration and Asylum'
+            when (l_jurisdiction='PUBLICLAW')  then 'Public Law'
+            else l_jurisdiction
+        end;
+
+    l_case_type_label =
+        case l_case_type_id IS NOT NULL
+           when (l_case_type_id='CIVIL')  then 'Civil'
+           when (l_case_type_id='PRLAPPS')  then 'Private Law'
+           when (l_case_type_id='PUBLICLAW')  then 'Public Law'
```

Comment:
I think `Asylum` is missing?
I don't know what the caseType is for PUBLICLAW, but we agreed to map it to Public Law.  Please check this.

Reply:
Taking values of case_type_id from Prod DB, looks like Asylum is already stored in the same format. So, it's covered as part of           " else l_case_type_id " and NO Translation is needed.

Reply:
And for PUBLICLAW, yes it's mapped to Public Law as per the description in ticket : https://tools.hmcts.net/jira/browse/RWA-2658